src/scripts/configure_npm_for_heroku.sh

```diff
@@ -0,0 +1,86 @@
+#!/bin/bash
+
+# shellcheck disable=SC2016
+# shellcheck disable=SC2129
+
+set +e
+# Change to working directory (defaults to .)
+cd "$WORKING_DIR" || exit
+
+# Check if npm is installed
+if ! command -v npm &> /dev/null
+then
+  echo "npm is not installed. Please install npm before running this script."
+  exit 1
+fi
+
+# Check if NPM_REGISTRY and NPM_AUTH_CONFIG are set
+if [ -z "$CLOUDSMITH_NPM_REGISTRY" ]
+then
+  echo "Unable to configure NPM. Env var CLOUDSMITH_NPM_REGISTRY is not defined. Please run the set_env_vars_for_npm command first."
+  exit 1
+fi
+if [ -z "$CLOUDSMITH_NPM_AUTH_CONFIG" ]
+then
+  echo "Unable to configure NPM. Env var CLOUDSMITH_NPM_AUTH_CONFIG is not defined. Please run the set_env_vars_for_npm command first."
+  exit 1
+fi
+
+# Check if Heroku CLI is installed
+if ! command -v heroku &> /dev/null
+then
+    echo "Heroku CLI could not be found. Please install it to proceed."
+    exit 1
+fi
+
+# Check if jq is installed
+if ! command -v jq &> /dev/null
+then
+    echo "jq could not be found. Please install it to proceed."
+    exit 1
+fi
+
+# Check if APP_NAME is set
+if [ -z "$APP_NAME" ]
+then
+  echo "Unable to configure NPM. Env var APP_NAME is not defined. Please specify it as a parameter in your circleci job."
+  exit 1
+fi
+
+# Set environment variables via Heroku CLI
+output=$(heroku config:set NPM_REGISTRY="registry=$CLOUDSMITH_NPM_REGISTRY" --app "$APP_NAME" 2>&1)
+heroku config:set NPM_TOKEN="$CLOUDSMITH_NPM_AUTH_CONFIG" --app "$APP_NAME" &>/dev/null # Silencing output as it contains an auth token
+
+# Check for error message in output
+if echo "$output" | grep -q "Error"; then
+    echo "Heroku app name is invalid. Please check your Heroku app name is a valid Heroku app."
+    exit 1
+fi
```

Comment:
Should we be checking the second "heroku config" set command?

Is that error message correct - is this just a check for the Heroku App name? Could this fail for another reason i.e the config set command fails?

Reply:
The error message is based on the error you get back from the heroku CLI if you try to do a config set with an invalid app-name. I only included it once as it will fail to set for the auth config but to so silently, so we only need to check it once and error.

I'm not sure why it would fail for any other reason